simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java

```diff
@@ -125,28 +125,73 @@ public void clear() {
   public Enumeration<Collector.MetricFamilySamples> metricFamilySamples() {
     return new MetricFamilySamplesEnumeration();
   }
+
+  public Enumeration<Collector.MetricFamilySamples> filteredMetricFamilySamples(Set<String> includedNames) {
+    return new MetricFamilySamplesEnumeration(includedNames);
+  }
+
   class MetricFamilySamplesEnumeration implements Enumeration<Collector.MetricFamilySamples> {
 
-    private final Iterator<Collector> collectorIter = collectors().iterator();
+    private final Iterator<Collector> collectorIter;
     private Iterator<Collector.MetricFamilySamples> metricFamilySamples;
     private Collector.MetricFamilySamples next;
+    private Set<String> includedNames;
 
-    MetricFamilySamplesEnumeration() {
+    MetricFamilySamplesEnumeration(Set<String> includedNames) {
+      this.includedNames = includedNames;
+      collectorIter = includedCollectorIterator(includedNames);
       findNextElement();
     }
 
-    private void findNextElement() {
-      if (metricFamilySamples != null && metricFamilySamples.hasNext()) {
-        next = metricFamilySamples.next();
+    private Iterator<Collector> includedCollectorIterator(Set<String> includedNames) {
+      if (includedNames.isEmpty()) {
+        return collectors().iterator();
       } else {
+        HashSet<Collector> collectors = new HashSet<Collector>();
+        synchronized (namesToCollectors) {
+          for(Map.Entry<String,Collector> entry :namesToCollectors.entrySet()){
```

Comment:
Formatting is still off, I'd suggest running something against all the files touched.